src/createClassProxy.js

```diff
@@ -75,10 +75,11 @@ function proxyClass(InitialComponent) {
     }
   }
 
+  let displayName = getDisplayName(InitialComponent);
   try {
     // Create a proxy constructor with matching name
     ProxyComponent = new Function('factory', 'instantiate',
-      `return function ${InitialComponent.name || 'ProxyComponent'}() {
+      `return function ${displayName}() {
```

Comment:
@gaearon this appears to cause an issue with Redux, it tries to evaluate `Connect(ComponentName)` as a function name and it results in a silent SyntaxError. It only shows up if you run with Pause On Caught Exceptions and no other error in console. Is it necessary to sanitize the function name?

This has worked before, not sure if it's a new problem with the latest change.